ports/stm32f4/mpconfigport.h

```diff
@@ -40,6 +40,7 @@
 #include "py/circuitpy_mpconfig.h"
 
 #define MICROPY_PORT_ROOT_POINTERS \
+    void *cpy_uart_obj_all[6]; \
```

Comment:
Can this be busio_uart_obj_t*? That way you won't need to cast it later.

Reply:
Unfortunately no - I cannot link peripheral files in portconfig without causing QSTR errors. Hence why I was wondering if there should maybe be an mcu-level config file outside of Peripherals, as with the boardconfig files. The current pattern across ports seems to be to just copy the same MCU information across all boards, which is kind of a maintainability hassle.

Reply:
QSTR errors can be a red herring because the deps on generating them isn't perfect. So, whenever you see an error with missing QSTR first try cleaning and rebuilding.

Please try it again with a clean build to see if you can use the real type. If it still doesn't work then please post the error.

Reply:
Consolidating all include discussion to here. Errors do occur on a clean build, though it did get rid of the QSTR (I remember getting some at one point on a clean build, so I'll link those if they show up again). Specific issue is that the micropython base objects are not defined at this point of compilation: 

```
In file included from ./peripherals/stm32f4/periph.h:34:0,
                 from ./mpconfigport.h:41,
                 from ../../py/mpconfig.h:45,
                 from ../../py/mpstate.h:31,
                 from ../../py/mpstate.c:27:
./peripherals/stm32f4/pins.h:40:5: error: unknown type name 'mp_obj_base_t'
     mp_obj_base_t base;
     ^~~~~~~~~~~~~
./peripherals/stm32f4/pins.h:64:14: error: unknown type name 'mp_obj_type_t'
 extern const mp_obj_type_t mcu_pin_type;
```

I could still use MAX_UART by having MCU specific defines in a new peripherals file, like mcuconfig.h, though I'd personally put that off until I have more than just 2 lines of defines. But I don't see any way to include full object types here.

Reply:
I think an MCU specific file is probably the easiest way to go. You want to add more to them later anyway right? Getting includes situated can be tricky.

Reply:
Ok, I'll draft something up for that. But I don't think we can avoid the void pointer, as noted.